Add Request.signal (#24465)
mdn/content: files/en-us/web/api/request/signal/index.md

---
title: Request.signal
slug: Web/API/Request/signal
page-type: web-api-instance-property
browser-compat: api.Request.signal
---

{{APIRef("Fetch API")}}

> The read-only **`signal`** property of the {{DOMxRef("Request")}} interface return the {{domxref("AbortSignal")}} associated with the request.

Reviewer: ```suggestion
The read-only **`signal`** property of the {{DOMxRef("Request")}} interface returns the {{domxref("AbortSignal")}} associated with the request.
```

## Value

An {{DOMxRef("AbortSignal")}} object.

## Examples

```js
// Create a new abort controller
const controller = new AbortController();

> // Creatae a request we this controller's AbortSignal object

Reviewer: ```suggestion
// Create a request with this controller's AbortSignal object
```

const req = new Request('/', { signal: controller.signal });

// Add an event handler logging a message in case of abort

> req.signal.addEventListener("signal",  () => { console.log('abort'); }

Reviewer 2: ```suggestion
req.signal.addEventListener("signal", () => {
  console.log("abort");
});
```
